expression/builtin/time.go

```diff
@@ -162,6 +162,21 @@ func builtinNow(args []interface{}, ctx map[interface{}]interface{}) (interface{
 	return t.RoundFrac(int(fsp))
 }
 
+func builtinDayName(args []interface{}, ctx map[interface{}]interface{}) (interface{}, error) {
+	v, err := builtinWeekDay(args, ctx)
+	if err != nil {
+		return nil, err
+	}
+	if types.IsNil(v) {
+		return nil, nil
+	}
+	weekday := v.(int64)
+	if !((weekday >= 0) && (weekday < int64(len(mysql.WeekdayNames)))) {
```

Comment:
I think it's better to rewrite the expression to 
`weekday < 0 || weekday >= int64(len(mysql.WeekdayNames))`